src/python/pants/backend/cc/lint/clangformat/subsystem.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+import os
+from typing import Iterable
+
+from pants.backend.python.goals import lockfile
+from pants.backend.python.goals.lockfile import GeneratePythonLockfile
+from pants.backend.python.subsystems.python_tool_base import PythonToolBase
+from pants.backend.python.subsystems.setup import PythonSetup
+from pants.backend.python.target_types import ConsoleScript
+from pants.core.goals.generate_lockfiles import GenerateToolLockfileSentinel
+from pants.core.util_rules.config_files import ConfigFilesRequest
+from pants.engine.rules import Rule, collect_rules, rule
+from pants.engine.unions import UnionRule
+from pants.option.option_types import ArgsListOption, SkipOption
+from pants.util.docutil import git_url
+from pants.util.strutil import softwrap
+
+
+class ClangFormat(PythonToolBase):
```

Comment:
We probably want an option to turn off config discovery:

https://github.com/pantsbuild/pants/blob/2ce2d46abc95d1af266068f98a9095ad5a912f24/src/python/pants/backend/python/lint/isort/subsystem.py#L62-L72

Reply:
Yep, I can do that - clang-format without the config would be ... weird... I have to say. Unless I expose the `style` option. But maybe I can do that down the road. I've never seen a C/C++ project that uses clang-format without the config. 

Right now, if they don't use the `.clang-format` file, it'll default to WebKit formatting without any complaining (the `fallback-style` option).

Reply:
Definitely not blocking if this will be hard to pull off. It's probably premature feature

Reply:
Going to hold back on this one until I can think of some better examples, or until I can figure out a way to tweak the API so it makes a bit more sense.